internal/controller/application_controller.go

```diff
@@ -282,19 +282,19 @@ func (r *ApplicationPullSecretCreator) createApplicationPullSecret(ctx context.C
 	return nil
 }
 
-// udateServiceAccountWithApplicationPullSecret updates the ServiceAccount to include
+// updateServiceAccountWithApplicationPullSecret updates the ServiceAccount to include
```

Comment:
The name is misleading and unclear.
What about `updateIntegrationServiceAccountWithNamespacePullSecret`?
But then why it happens in the application controller? Do we need to even have this controller?